R/groups.R

```diff
@@ -0,0 +1,188 @@
+#'
+#' @return
+#' A tibble with the following columns:
+#'
+#'   * `guid`: The unique identifier of the group
+#'   * `name`: The group name
+#'   * `owner_guid`: The group owner's unique identifier. When using LDAP or
+#'     Proxied authentication with group provisioning enabled this property
+#'     will always be null.
+#'
+#' @details
+#' Please see https://docs.posit.co/connect/api/#get-/v1/groups for more information.
+#'
+#' @examples
+#' \dontrun{
+#' library(connectapi)
+#' client <- connect()
+#'
+#' # get all groups
+#' get_groups(client, limit = Inf)
+#' }
+#'
+#' @family groups functions
+#' @export
+get_groups <- function(src, page_size = 500, prefix = NULL, limit = Inf) {
+  validate_R6_class(src, "Connect")
+
+  # The `v1/groups` endpoint always returns the first page when `prefix` is
+  # specified, so the page_offset function, which increments until it hits an
+  # empty page, fails.
+  if (!is.null(prefix)) {
+    response <- src$groups(page_size = page_size, prefix = prefix)
+    res <- response$results
+  } else {
+    res <- page_offset(src, src$groups(page_size = page_size, prefix = NULL), limit = limit)
+  }
+
+  out <- parse_connectapi_typed(res, connectapi_ptypes$groups)
+
+  return(out)
+}
+
+#' Get users within a specific group
+#'
+#' @param src A Connect client object
+#' @param guid A group GUID identifier
+#'
+#' @return
+#' A tibble with the following columns:
+#'
+#'   * `email`: The user's email
+#'   * `username`: The user's username
+#'   * `first_name`: The user's first name
+#'   * `last_name`: The user's last name
+#'   * `user_role`: The user's role. It may have a value of administrator,
+#'     publisher or viewer.
+#'   * `created_time`: The timestamp (in RFC3339 format) when the user
+#'     was created in the Posit Connect server
+#'   * `updated_time`: The timestamp (in RFC3339 format) when the user
+#'     was last updated in the Posit Connect server
+#'   * `active_time`: The timestamp (in RFC3339 format) when the user
+#'     was last active on the Posit Connect server
+#'   * `confirmed`: When false, the created user must confirm their
+#'     account through an email. This feature is unique to password
+#'     authentication.
+#'   * `locked`: Whether or not the user is locked
+#'   * `guid`: The user's GUID, or unique identifier, in UUID RFC4122 format
+#'
+#' @details
+#' Please see https://docs.posit.co/connect/api/#get-/v1/groups/-group_guid-/members
+#' for more information.
+#'
+#' @examples
+#' \dontrun{
+#' library(connectapi)
+#' client <- connect()
+#'
+#' # get the first 20 groups
+#' groups <- get_groups(client)
+#'
+#' group_guid <- groups$guid[1]
+#'
+#' get_group_members(client, guid = group_guid)
+#' }
+#'
+#' @family groups functions
+#' @export
+get_group_members <- function(src, guid) {
+  validate_R6_class(src, "Connect")
+
+  res <- src$group_members(guid)
+
+  out <- parse_connectapi(res$results)
+
+  return(out)
```

Comment:
```suggestion
  parse_connectapi(res$results)
```

Two things: (1) we don't need to write this to `out` yeah? (2) We use implicit returns at the end of functions, right?